geopandas/tools/overlay.py

```diff
@@ -176,3 +177,158 @@ def overlay(df1, df2, how, use_sindex=True):
 
     # Return geodataframe with new indices
     return GeoDataFrame(collection, index=range(len(collection)))
+
+def overlay_intersection(df1, df2):
+    ''''
+    Overlay Intersection operation used in overlay function
+    '''
+    # Spatial Index to create intersections
+    spatial_index = df2.sindex
+    df1['bbox'] = df1.geometry.apply(lambda x: x.bounds)
```

Comment:
I think you can assign this to a temporary variable instead of assigning it into the dataframe. That also makes that you don't have to remove that column again later on